src/sensai/evaluation/crossval.py

```diff
@@ -86,6 +86,7 @@ def __init__(self, data: InputOutputData, folds: int = 5, randomSeed=42, returnT
             testIndices = permutedIndices[testStartIdx:testEndIdx]
             trainIndices = np.concatenate((permutedIndices[:testStartIdx], permutedIndices[testEndIdx:]))
             self.modelEvaluators.append(self._createModelEvaluator(data.filterIndices(trainIndices), data.filterIndices(testIndices)))
+        super().__init__()
```

Comment:
Calling the super constructor should be the first (not the last) thing we do whenever possible.

Reply:
I have a proposal of using a mixing for setting experiments, it's in a separate commit that can be dropped if you don't like the mechanism. It makes the code simpler, no need for inits and duplications. It also works as one would expect, since assigning a class attribute from an instance actually changes the class attribute to the instance. Only mutation of class attributes gives strange behaviour but this is not possible since None is unmutable.

The downside is, that several classes now have a class attribute `trackedExperiment` which evaluates to None. I would say it is only an aesthetical issue and I am personally not bothered by it. What do you think? Dropping the last commit is easy

Reply:
If you don't like the public class attribute, I could also make it private and add a public property getter in the mixin

Reply:
The implementation is somewhat "dangerous", as it does not follow the conventions of setters. A caller will assume that calling `setTrackedExperiment` on one instance has no impact on any other instance. This is not the case with the current implementation, as we are sharing static state here. In particular, it would be rather unexpected for a user to see that class A's executions are being tracked only because an instance of unrelated class B had an experiment set (as a result of class A also "being" a TrackingMixin). Furthermore, it seems limited that we can only ever have one tracked experiment at any given time; it does not seem far-fetched to want to simultaneously track several parallel experiments.

Also: I am curious what will happen in a multi-processing context here: Will a static attribute of the object being serialised carry over to the new process? I really don't know what will happen. Unless multiprocessing specifically considers this, it won't, as serialising the actual parameters of the functions being "multi-processed" will not consider static attributes.

Reply:
Nice, the best of both worlds! I'll change it accordingly, although it will still remain somehow hacky, I guess

Reply:
Done. Although it does not really change anything, right? If the instance attribute is not set, if will return the class attribute which was None

Reply:
It improves the code quality and requires a less "sharp eye": If you see class attribute `_foo` and you see `self._foo = bar`, you assume that the class attribute is being set (even though it's not).
But functionally, yes, it's the same.

Reply:
Your update didn't remove the class attribute. It's not required anymore.

Reply:
Done. Even before reading your second comment ^^.  Pycharm complains now but we have to let it complain

Reply:
If you want to avoid the warnings, always use `setattr` and `getattr`, ideally with a class-level attribute for the name.

Reply:
Done. I wouldn't necessarily say it's much cleaner but the warnings are gone

Reply:
Importantly, it's significantly cleaner than the shadowed attribute, which is the main thing to improve upon.

Reply:
Absolutely! I meant that `setattr` was not much cleaner than doing `self._trackedExperiment = ...`. Removing the class-attribute was definitely worth it!